CMakeLists.txt

```diff
@@ -142,6 +142,14 @@ else()
   assert(0 "Unreachable code. Expected optimization level to have been chosen. See Kconfig.zephyr")
 endif()
 
+if(NOT (CONFIG_ARC OR CONFIG_ARM OR CONFIG_X86 OR CONFIG_NIOS2 OR CONFIG_RISCV OR
```

Comment:
I guess there is no way to make it more systematic, but this is likely to be missed in a next architecture porting.